lib/libvfio-user.c

```diff
@@ -1976,19 +1993,30 @@ vfu_sg_is_mappable(vfu_ctx_t *vfu_ctx, dma_sg_t *sg)
     return dma_sg_is_mappable(vfu_ctx->dma, sg);
 }
 
-EXPORT void
-vfu_migr_done(vfu_ctx_t *vfu_ctx, int reply_errno)
+EXPORT int
+vfu_async_done(vfu_ctx_t *vfu_ctx, int reply_errno)
 {
+    int ret = 0;
+
     assert(vfu_ctx != NULL);
-    assert(vfu_ctx->migr_trans_pending);
+    assert(vfu_ctx->pending != VFU_CTX_PENDING_NONE);
 
-    if (!vfu_ctx->migr_trans_msg->hdr.flags.no_reply) {
-        do_reply(vfu_ctx, vfu_ctx->migr_trans_msg, reply_errno);
+    if (vfu_ctx->pending == VFU_CTX_PENDING_DMA_UNMAP) {
```

Comment:
what about map? isn't the point of async map that we don't touch the DMA region array until consumer tells us?